src/database.ts

```diff
@@ -1388,6 +1536,90 @@ class Database extends GrpcServiceObject {
     });
     return waitForSessionStream;
   }
+
+  restore(backupPath: string): Promise<RestoreDatabaseResponse>;
+  restore(
+    backupPath: string,
+    options?: CallOptions
+  ): Promise<RestoreDatabaseResponse>;
+  restore(backupPath: string, callback: RestoreDatabaseCallback): void;
+  restore(
+    backupPath: string,
+    options: CallOptions,
+    callback: RestoreDatabaseCallback
+  ): void;
+  /**
+   * @typedef {array} RestoreDatabaseResponse
+   * @property {Database} 0 The new {@link Database}.
+   * @property {Operation} 1 An {@link Operation} object that can be used to check
+   *     the status of the request.
+   * @property {object} 2 The full API response.
+   */
+  /**
+   * @callback RestoreDatabaseCallback
+   * @param {?Error} err Request error, if any.
+   * @param {Database} database The new {@link Database}.
```

Comment:
Let's not return the Database instance, since they already have it.

Reply:
I wanted to clarify something here. `RestoreDatabaseCallback` is a `LongRunningCallback`.

https://github.com/googleapis/nodejs-spanner/blob/8d4a32e0a9fb36435e9c3caf2bec0919b29f2984/src/common.ts#L57

This expects us to return the Resource which in this case is Database. Should I change this to `IDatabase` and return that instead or change to a different callback type? Possibly will have to create a custom one.

Reply:
Ah, thanks. I think this was an accident somewhere along the way. I see that interface currently being used where a parent method creates a child, so it is required to pass that instance back to the user, e.g. `instance.createDatabase((err, database, operation, apiResponse) => {})`. However, "database.create()" shouldn't be returning "database", since in this case, we are creating the  object the user already has, but it is.

No big deal, though. Consistent is good! Sorry for the confusion.